### PR TITLE
Add RGD & Remove interceptors /1

### LINKED ISSUE #30381: Remove "Interceptors" heading and content from native AoT docs
We can remove the [C# 12 Interceptors](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/native-aot?view=aspnetcore-8.0#c-12-interceptors) from the native AoT docs. We should document the request delegate generator on a separate page (alongside the diagnostics it emits) and outline the dependency on the interceptors feature there.

cc: @Rick-Anderson 

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 8839fa1e-7c10-4b22-d03d-669906a2b9ea
* Version Independent ID: 8839fa1e-7c10-4b22-d03d-669906a2b9ea
* Content: [ASP.NET Core support for native AOT](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/native-aot?view=aspnetcore-8.0)
* Content Source: [aspnetcore/fundamentals/native-aot.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/fundamentals/native-aot.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @mitchdenny
* Microsoft Alias: **midenn**

---
[Associated WorkItem - 161535](https://dev.azure.com/msft-skilling/Content/_workitems/edit/161535)

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #30381

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/fundamentals/aot/request-delegate-generator/rdg.md](https://github.com/dotnet/AspNetCore.Docs/blob/d1912234c92c505131491760a161d90a0918ac2d/aspnetcore/fundamentals/aot/request-delegate-generator/rdg.md) | [aspnetcore/fundamentals/aot/request-delegate-generator/rdg](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/aot/request-delegate-generator/rdg?branch=pr-en-us-30414) |
| [aspnetcore/fundamentals/native-aot.md](https://github.com/dotnet/AspNetCore.Docs/blob/d1912234c92c505131491760a161d90a0918ac2d/aspnetcore/fundamentals/native-aot.md) | [ASP.NET Core support for native AOT](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/native-aot?branch=pr-en-us-30414) |

### CODE REVIEW

File: aspnetcore/fundamentals/native-aot.md [hunk 1]
```diff
@@ -248,9 +248,9 @@ A parameter on the delegate that isn't bound to the body and does ***not*** need
 
 :::code language="csharp" source="~/fundamentals/aot/samples/Todo.cs" id="snippet_1":::
 
-### C# 12 Interceptors
+<!-- ### C# 12 Interceptors
 
-[C# 12 interceptors compiler feature](/dotnet/csharp/whats-new/csharp-12) to support intercepting calls to minimal API’s [Map](xref:Microsoft.AspNetCore.Builder.EndpointRouteBuilderExtensions). Map method calls are replaced with code generated calls at compile time. The compiler generated implementations are compatible with native AOT and results in increased startup performance for apps published with the `PublishAot` property enabled.
+[C# 12 interceptors compiler feature](/dotnet/csharp/whats-new/csharp-12) supports intercepting calls to minimal API’s [Map](xref:Microsoft.AspNetCore.Builder.EndpointRouteBuilderExtensions). `Map` method calls are replaced with code generated calls at compile time. The compiler generated implementations are compatible with native AOT and results in increased startup performance for apps published with the `PublishAot` property enabled. -->
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/native-aot.md:253]: Commented out in case we ever want to use this information. Let me know if I should just delete it.

**Comment by Rick-Anderson** [comment]
@captainsafia  I'll update this doc once #30423 merges.

**Comment by captainsafia** [comment]
@Rick-Anderson Is it possible to get the new pages to surface under the `Native AoT` section in the docs?

image

File: aspnetcore/fundamentals/aot/request-delegate-generator/rdg.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+content_well_notification: AI-contribution
+monikerRange: '>= aspnetcore-8.0'
+ms.topic: article
+ms.prod: aspnet-core
+ms.date: 9/21/2023
+uid: fundamentals/aot/rdg
+---
+# Turn Map methods into request delegates with the ASP.NET Core Request Delegate Generator
+
+The ASP.NET Core Request Delegate Generator (RDG) is a tool that generates request delegates for ASP.NET Core apps. The RDG is used by the native ahead-of-time (AOT) compiler to generate request delegates for the app's `Map` methods.
```
Comment (captainsafia) [inline aspnetcore/fundamentals/aot/request-delegate-generator/rdg.md:16]:
```suggestion
The ASP.NET Core Request Delegate Generator (RDG) is a compile-time source generator that compiles route handlers provided to a minimal API to request delegates that can be processed by ASP.NET Core's routing infrastructure. The RDG is implicitly enabled when applications are published with AoT enabled and generated trim and native AoT-friendly code.
```

File: aspnetcore/fundamentals/native-aot.md (hunk 1 above)
Comment (captainsafia) [reply aspnetcore/fundamentals/native-aot.md:253]: It's OK to delete this.

**Review by captainsafia — APPROVED** [review]
LGTM

**Comment by Rick-Anderson** [comment]
> @Rick-Anderson Is it possible to get the new pages to surface under the `Native AoT` section in the docs?
> 
> image

@captainsafia  it's live now. We don't like to have the toc file long in a PR to avoid merge conflicts. We typically add entries right after a new topic PR merges.

image

**Pull request merged**